cvmfs/catalog_mgr_ro.cc

```diff
@@ -14,43 +14,56 @@ using namespace std;  // NOLINT
 
 namespace catalog {
 
+// TODO(herethebedragons) correct return value and root_ctlg_location?
+LoadReturn SimpleCatalogManager::GetNewRootCatalogContext(
+                                                       CatalogContext *result) {
+  if (result->hash().IsNull()) {
+    result->SetHash(base_hash_);
+  }
+  result->SetRootCtlgLocation(kCtlgLocationServer);
+  result->SetMountpoint(PathString("", 0));
+
+  return kLoadNew;
+}
+
+
 /**
  * Loads a catalog via HTTP from Statum 0 into a temporary file.
- * @param url_path the url of the catalog to load
- * @param mount_point the file system path where the catalog should be mounted
- * @param catalog_file a pointer to the string containing the full qualified
- *                     name of the catalog afterwards
- * @return 0 on success, different otherwise
+ * See CatalogContext class description for correct usage
+ * 
+ * @return kLoadNew on success
  */
-LoadError SimpleCatalogManager::LoadCatalog(const PathString  &mountpoint,
-                                            const shash::Any  &hash,
-                                            std::string       *catalog_path,
-                                            shash::Any        *catalog_hash)
-{
-  shash::Any effective_hash = hash.IsNull() ? base_hash_ : hash;
+// TODO(herethebedragons) CORRECT?
+LoadReturn SimpleCatalogManager::LoadCatalogByHash(
+                                                 CatalogContext *ctlg_context) {
+  const shash::Any effective_hash = ctlg_context->hash();
   assert(shash::kSuffixCatalog == effective_hash.suffix);
   const string url = stratum0_ + "/data/" + effective_hash.MakePath();
 
-  FILE *fcatalog = CreateTempFile(dir_temp_ + "/catalog", 0666, "w",
-                                  catalog_path);
+  std::string tmp;
+
+  FILE *fcatalog = CreateTempFile(dir_temp_ + "/catalog", 0666, "w", &tmp);
+  ctlg_context->GetSqlitePathPtr()->assign(tmp);
```

Comment:
Also: let's move to after the `if` block
```suggestion
  ctlg_context->SetSqlitePath(tmp);
```